manticore/models/linux.py

```diff
@@ -1288,6 +1288,52 @@ def sys_getegid(self, cpu):
         '''
         return 1000
 
+    def sys_readv(self, cpu, fd, iov, count):
+        '''
+        Works just like C{sys_read} except that data is read into multiple buffers (for Linux 64 bits).
+        :rtype: int
+
+        :param cpu: current CPU.
+        :param fd: the file descriptor of the file to read.
+        :param iov: the buffer where the the bytes to read are stored.
+        :param count: amount of C{iov} buffers to read from the file.
+        :return: the amount of bytes read in total.
+        '''
+        total = 0
+        for i in xrange(0, count):
+            buf = cpu.read_int(iov + i * 16, 64)
+            size = cpu.read_int(iov + i * 16 + 8, 64)
+
+            data = self.files[fd].read(size)
+            total += len(data)
+            cpu.write_bytes(buf, data)
+            self.syscall_trace.append(("_read", fd, data))
+            logger.debug("READV(%r, %r, %r) -> <%r> (size:%r)"%(fd, buf, size, data, len(data)))
+        return total
+
+    def sys_readv32(self, cpu, fd, iov, count):
+        '''
+        Works just like C{sys_read} except that data is read into multiple buffers (for Linux 32 bits).
+        :rtype: int
+
+        :param cpu: current CPU.
+        :param fd: the file descriptor of the file to read.
+        :param iov: the buffer where the the bytes to read are stored.
+        :param count: amount of C{iov} buffers to read from the file.
+        :return: the amount of bytes read in total.
+        '''
+        total = 0
+        for i in xrange(0, count):
+            buf = cpu.read_int(iov + i * 16, 32)
```

Comment:
Oops! Yes I definitely agree about reducing copy+pasted code. While I'm at it, I can also fix sys_writev. What about using cpu.address_bit_size though? It should also be possible to write a method that takes the bitness as argument and returns an implementation. Would probably look like this in the syscall table: `0xXXXX: self.sys_readv(bits=32)`